homework/src/index.js

```diff
@@ -1,47 +1,119 @@
+          let fetchToDo = contributors.length;
+          contributors.forEach(contributor => {
+            fetchJSON(contributor.url, (err, user) => {
+              Object.assign(contributor, user);
+              --fetchToDo;
+              0 === fetchToDo &&
+                (console.log(contributors),
+                  contributors.forEach(contrUser => {
+                    let sectRcontrs = crnap("div", sectR, { id: "contributors" });
+                    sectRcontrs = crnap("div", sectRcontrs, { class: "card" });
+                    let secRimglink = crnap("a", sectRcontrs, { href: contrUser.html_url });
+                    secRimglink = crnap("figure", secRimglink, {});
+                    crnap("img", secRimglink, { src: contrUser.avatar_url });
+                    crnap("h2", sectRcontrs, { html: null == contrUser.name ? "" : `name: ${contrUser.name}` });
+                    crnap("p", sectRcontrs, { html: null == contrUser.login ? "" : `nickname: ${contrUser.login}` });
+                    crnap("p", sectRcontrs, {
+                      html: null == contrUser.location ? "" : `location: ${contrUser.location}`
+                    });
+                    crnap("p", sectRcontrs, {
+                      html: null == contrUser.followers ? "" : `followers: ${contrUser.followers}`
+                    });
+                    crnap("p", sectRcontrs, {
+                      html: null == contrUser.company ? "" : `company: ${contrUser.company}`
+                    });
+                    crnap("p", sectRcontrs, {
+                      html: null == contrUser.contributions ? "" : `contributions: ${contrUser.contributions}`
+                    });
+                    crnap("p", sectRcontrs, {
+                      html:
+                        null == contrUser.updated_at
+                          ? ""
+                          : `last github visit: ${new Date(contrUser.updated_at).toLocaleString()}`
+                    });
+                  }));
+            });
+          });
+        });
       }
-    };
-    xhr.onerror = () => cb(new Error('Network request failed'));
-    xhr.send();
+      const root = document.getElementById("root");
+      let header = crnap("header", root, {});
+      header = crnap("div", header, { class: "selectfield" });
+      const menu = crnap("select", header, { id: "selectmenu" });
+      var main = crnap("main", root, { id: "main" });
+      err
+        ? crnap("div", root, { a: err.message, b: "alert-error" })
+        : (data
+          .sort((b, a) => b.name.localeCompare(a.name))
+          .forEach((a, d) =>
+            crnap("option", menu, { value: d, b: "select", html: a.name })
+          ),
+          menu.addEventListener("change", event => {
+            event = event.target.value;
+            rend(data[event]);
+          }));
+    });
   }
-
-  function createAndAppend(name, parent, options = {}) {
+  function crnap(name, prnt, opts) {
```

Comment:
`crnap` is crappy name 😉. Also, separate function definitions with a blank line.